README.md

```diff
@@ -1,84 +1,103 @@
 # source-configs
-Provides structure to define app-level configuration schemas and automatically consume values via multiple sources in Node.js applications.
+A Node.js module that provides structure to define app-level configuration schemas and automatically consume values via multiple sources in Node.js applications. This module was built and is maintained by the [Roosevelt web framework team](https://github.com/rooseveltframework/roosevelt), but it can be used independently of Roosevelt as well. 
 
-## Install
+Configs matching a given schema are sourced from the following locations in the following order of precedence:
 
-You can grab source-configs from NPM with NPM or yarn:
+- Command line argument set in the schema via `commandLineArg`.
+- Environment variable set in the schema via `envVar`.
+- Deployment config file declared via:
+  - Command line argument: `--deploy-config` or `--dc`.
+  - Environment variable: `DEPLOY_CONFIG`.
+  - package.json as a `deployConfig` field.
+- Default defined in the schema.
 
-```
-npm install source-configs
-```
+## Usage
 
-or
+First declare `source-configs` as a dependency in your app.
 
-```
-yarn add source-configs
+Then create a schema object.
+
+A schema is an object that serves as a template for the desired final configuration consumed by your app.
+
+Here's a simple example declaring that our application will use a WebSocket with a configurable host, port, and protocol: 
+
+```js
+{
+  websocket: {
+    host: {},
+    port: {},
+    protocol: {}
+  }
+}
 ```
 
-## Priority basis
+### Schema metadata
 
-SourceConfigs will examine the 4 following locations and will fallback to the one below it if it doesn't find the config value
+Schemas support the following metadata for each configurable property in order to create constraints if desired or add additional functionality:
 
-* Command line arguments set in the schema under commandLineArg
-* Environment variable set in the schema as envVar
-* Deployment config file declared in the package.json as a `deployConfig` field, as a command line argument of `--deploy-file` or `--df`, or an environment variable of `SC_DEPLOY_FILE`.
-* Default defined in the schema
+- `description` *[String]*: Describe what this config will be used for.
+- `default` *[any]*: Set a default value for this config. If not set, the default is null.
+- `values` *[Array]*: Enumerated list of values that are valid. If not set, any value will be valid.
+- `commandLineArg` *[String]*: Command line argument to set this config. Will be converted to camelCase on the Node.js side. Example: the command line argument `--doThing` would be `doThing` on the Node.js side. If not set, source-configs will not listen for command line arguments to set the value for this config. [TODO: is the camelCaser really desirable?]
```

Comment:
Both yargs-parser and the command line flags parser in Roosevelt convert the flags into a flat map where the flags are camelCased. Is there a counterexample or opinion against having the field be camelCased?

Reply:
Okay, actually, after examining this a bit more, explicitly describing the command line argument here as it would be written when running the command may be easier to understand. Resolving this.